docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,656 @@
+    to: "0x291B0E3aA139b2bC9Ebd92168575b5c6bAD5236C",
+    tokenAddress: "0xad50f302a957C165d865eD398fC3ca5A5A2cDA85",
+    chain: "baseSepolia",
+    amount: "4",
+    decimals: 18,
+    chainId: 84532,
+};
+
+const chainBParams = {
+    from: "0x291B0E3aA139b2bC9Ebd92168575b5c6bAD5236C",
+    to: "0x48e6a467852Fa29710AaaCDB275F85db4Fa420eB",
+    tokenAddress: "0x2dcA1a80c89c81C37efa7b401e2d20c1ED99C72F",
+    chain: "yellowstone",
+    amount: "8",
+    decimals: 18,
+    chainId: 175188,
+};
+/* 
+deposit1: wallet A deposits on chain B, if action executes, funds are transferred to wallet B
+deposit2: wallet B deposits on chain A, if action executes, funds are transferred to wallet A
+*/
+```
+:::info
+On the chains section, we can specify any of the chains [supported](../resources/supported-chains) by LIT.
+:::
+
+You can find a generator which takes the above params to generate a Lit Action for swap: [swapActionGenerator.js](https://github.com/anshss/lit-evm-swap/blob/main/lit/swapActionGenerator.js)
+
+Remember, once a Mint/Grant/Burn PKP is created, anyone can execute the Lit Action on the Lit Network. This execution needs to define the parameters needed by the Lit Action to run. While generating a sign or checking conditions on the action, we need information around the pkp for which we are doing it. Due to this, we need to construct our Lit Action in a way that works with any PKP.
+
+Our Lit Action primarily focuses on three factors:
+
+- Access Conditions for checking funds on both chains
+- Transaction objects for transferring funds between swap parties
+- Clawback Transaction objects for revert transferring funds to their owners
+- Lit Action conditions which decide which transaction object to sign and return
+
+We'll use [Lit's Access Control](../sdk/access-control/evm/basic-examples#must-posess-at-least-one-erc20-token) here to verify if the conditions are being met for the swap. There need to be 2 conditions for a swap to execute,
+
+- Have token A funds reached on Chain A?
+- Have token B funds reached on Chain B?
+
+If yes, then forward the funds and complete the swap or else revert them.
+
+A condition on Base Sepolia for checking if the PKP contains the exact amount of token A in Wei will look like this:
+
+```js
+const chainACondition = {
+    conditionType: "evmBasic",
+    contractAddress: "0xad50f302a957C165d865eD398fC3ca5A5A2cDA85",
+    standardContractType: "ERC20",
+    chain: "baseSepolia",
+    method: "balanceOf",
+    parameters: ["address"],
+    returnValueTest: { comparator: "==", value: "4000000000000000000" },
+};
+// parameter field would be later replaced by the pkpAddress in the action
+```
+
+There need to be two transaction objects, each of which transfers funds between the swap parties. Each of them will be calling function `transfer(address, uint256)` on the token's smart contract. This information is captured in the "data" field of the object. We can get that in the following way,
```

Comment:
> There need to be two transaction objects, each of which transfers funds between the swap parties. Each of them will be calling function `transfer(address, uint256)` on the token's smart contract. This information is captured in the "data" field of the object. We can get that in the following way,

"We need to create two transaction objects, each of which will transfer the funds between the two swap parties that have been set. Each party will call the `transfer(address, uint256)` function on the token's smart contract. This information is captured in the "data" field of the object which we can fetch in the following way:"